railties/lib/rails/application/configuration.rb

```diff
@@ -283,6 +283,10 @@ def load_defaults(target_version)
           if respond_to?(:action_controller)
             action_controller.allow_deprecated_parameters_hash_equality = false
           end
+
+          if respond_to?(:active_record)
+            active_record.use_foreign_keys = true
```

Comment:
Might make more sense to have this as an adapter configuration, no?

E.g we'd pass `use_foreign_key: false` in the `database.yml`, and that would cause `supports_foreign_keys?` to return `false`? That wouldn't change any API, and allow to configure this on a per DB basis.